### PR TITLE
Ch16 summary, exercises, and go further blocks

### PR DESCRIPTION
**Pull request by pavpanchekha**
This PR does a pass on the summary, exercises, and go further blocks. The go further blocks are done (one per section).

### CODE REVIEW

File: book/invalidation.md [hunk 1]
```diff
@@ -1427,9 +1427,21 @@ right now, because the `BlockLayout` assumes its children's `height`
 fields are protected, but if those fields are `LineLayout`s they aren't.
 
 ::: {.further}
-TODO
+
+Dirty flags aren't the only way to achieve incremental performance;
+another option is to keep track of *delta*s. For example, in the
+[Adapton][adapton] project, each computation that converts inputs to
+outputs can also convert input changes to output changes. [Operational
```
Comment (chrishtr) [inline book/invalidation.md:1434]:
```suggestion
outputs can also convert input changes to output changes (i.e. deltas). [Operational
```

File: book/invalidation.md [hunk 2]
```diff
@@ -2254,10 +2296,26 @@ exercise.] is changed, it won't invalidate any layout fields (because
 these properties don't affect any layout fields) and so animations will
 once again skip layout entirely.
 
+
 ::: {.further}
-TODO
+
+CSS style depend on which elements a selector matches, and as the page
```
Comment (chrishtr) [inline book/invalidation.md:2302]:
```suggestion
CSS styles depend on which elements a selector matches, and as the page
```

File: book/invalidation.md [hunk 3]
```diff
@@ -2254,10 +2296,26 @@ exercise.] is changed, it won't invalidate any layout fields (because
...
+make invalidation [more complicated][has-invalidation], but this
+complexity is necessary for fast re-styles.
+
 :::
 
+[invalidation-set]: https://chromium.googlesource.com/chromium/src/+/HEAD/third_party/blink/renderer/core/css/style-invalidation.md?pli=1#
+[has-invalidation]: https://blogs.igalia.com/blee/posts/2023/05/31/how-blink-invalidates-styles-when-has-in-use.html
+[^our-browser]: The reason this isn't necessary in our browser is just
+    that our browser supports too few CSS selectors and too few DOM
+    APIs for any selector matches to actually change!
```
Comment (chrishtr) [inline book/invalidation.md:2317]: There is `:focus` and innerHTML, which do cause this. I'd say instead that our browser has too few features for such advanced techniques to make sense.

File: book/invalidation.md [hunk 4]
```diff
@@ -2539,54 +2597,41 @@ I've left exploring it to an advanced exercise.
...
+- A good browser aims for the principle of incremental performance:
+  the cost of an change should be proportional to size of the change,
+  not the size of the page as a whole.
 
-* Cache invalidation is difficult and error-prone, and justifies careful
-work and good abstractions like `ProtectedField`.
+- Cache invalidation is difficult and error-prone,
+  and justifies careful abstractions like `ProtectedField`.
+
+- Both data dependencies and control dependencies can be invalidated
```
Comment (chrishtr) [inline book/invalidation.md:2633]: "control dependencies" were not introduced as a term of art in the latest draft of this chapter. Maybe rephrase to "data dependencies and tree walk dependencies"?

Comment (pavpanchekha) [reply book/invalidation.md:2633]: Rewrote to "Invalidation can be used to skip allocation, computation, or even traversals."

**Review by chrishtr — APPROVED** [review]
